hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/helpers/DatanodeIdYaml.java

```diff
@@ -0,0 +1,186 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.ozone.container.common.helpers;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.collections.MapUtils;
+import org.apache.hadoop.hdds.protocol.DatanodeDetails;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.Yaml;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * Class for creating datanode.id file in yaml format.
+ */
+public class DatanodeIdYaml {
+
+  private DatanodeIdYaml() {
+
+  }
+
+  /**
+   * Creates a yaml file using DatnodeDetails. This method expects the path
+   * validation to be performed by the caller.
+   *
+   * @param datanodeDetails {@link DatanodeDetails}
+   * @param path            Path to datnode.id file
+   */
+  public static void createDatanodeIdFile(DatanodeDetails datanodeDetails,
+                                          File path) throws IOException {
+    Preconditions.checkNotNull(path);
```

Comment:
This is usually an anti-pattern, since we will crash below with NPE if path was somehow null.

Reply:
Thanks will remove it from write path.